Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -573,11 +620,74 @@ private ProjectRootElement CreateValidProjectSetup(WellKnownProjectProperties pr
 
             var projectRoot = BuildUtilities.CreateValidProjectRoot(TestContext, projectFolder, projectProperties);
 
+            AddCaptureTargetsImport(projectFolder, projectRoot);
+
             var projectFilePath = Path.Combine(projectFolder, "valid.project.proj");
             projectRoot.Save(projectFilePath);
             return projectRoot;
         }
 
+        private void AddCaptureTargetsImport(string projectFolder, ProjectRootElement projectRoot)
+        {
+            // Add an additional import that will dump data we are interested in to the build log
+            var captureTargetsFilePath = CreateCaptureDataTargetsFile(projectFolder);
+            var captureTarget = projectRoot.AddImport(captureTargetsFilePath);
+
+            var projectFilePath = Path.Combine(projectFolder, "valid.project.proj");
+            projectRoot.Save(projectFilePath);
+        }
+
+        private string CreateCaptureDataTargetsFile(string directory)
+        {
+            // Most of the tests above want to check the value of build property
+            // or item group after a target has been executed. However, this
+            // information is not available through the buildlogger interface.
+            // So, we'll add a special target that writes the properties/items
+            // we are interested in to the message log.
+            // The SimpleXmlLogger has special handling to extract the data
+            // from the message and add it to the BuildLog.
```

Comment:
This is pretty ugly, but I couldn't think of a simpler way to get the information back from the out-of-proc build.